esmvaltool/diag_scripts/hyint/hyint_functions.R

```diff
@@ -0,0 +1,1341 @@
+                         colorbar.label = "image.scale", extend = T,
+                         line.label = 2, line.colorbar = 0, cex.label = 1,
+                         cex.colorbar = 1, colorbar.width = 1,
+                         new_fig_scale = c(-0.07, -0.03, 0.1, -0.1), ...) {
+
+  # save properties from main plotting region
+  old.par <- par(no.readonly = TRUE)
+  mfg.save <- par()$mfg
+  old.fig <- par()$fig
+
+  # defining plotting region with proper scaling
+  xscal <- (old.fig[2] - old.fig[1])
+  yscal <- (old.fig[4] - old.fig[3])
+  lw <- colorbar.width
+  lp <- line.colorbar / 100
+  new.fig <- c(old.fig[2] + new_fig_scale[1] * xscal * lw - lp,
+               old.fig[2] + new_fig_scale[2] * xscal - lp,
+               old.fig[3] + new_fig_scale[3] * yscal,
+               old.fig[4] + new_fig_scale[4] * yscal)
+
+  if (missing(levels)) {
+    levels <- seq(min(z), max(z), , 12)
+  }
+  # fixing color palette
+  col <- color.palette(length(levels) - 1)
+
+  # starting plot
+  par(mar = c(1, 1, 1, 1), fig = new.fig, new = TRUE)
+
+  # creating polygons for legend
+  poly <- vector(mode = "list", length(col))
+  for (i in seq(poly)) {
+    poly[[i]] <- c(levels[i], levels[i + 1], levels[i + 1], levels[i])
+  }
+
+  xlim <- c(0, 1)
+  if (extend) {
+    longer <- 1.5
+    dl <- diff(levels)[1] * longer
+    ylim <- c(min(levels) - dl, max(levels) + dl)
+  } else {
+    ylim <- range(levels)
+  }
+  plot(1, 1,
+    t = "n", ylim = ylim, xlim = xlim, axes = FALSE, xlab = "",
+    ylab = "", xaxs = "i", yaxs = "i", ...
+  )
+  for (i in seq(poly)) {
+    polygon(c(0, 0, 1, 1), poly[[i]], col = col[i], border = NA)
+  }
+  if (extend) {
+    polygon(c(0, 1, 1 / 2), c(levels[1], levels[1], levels[1] - dl),
+      col = col[1], border = NA
+    )
+    polygon(c(0, 1, 1 / 2), c(
+      levels[length(levels)], levels[length(levels)],
+      levels[length(levels)] + dl
+    ),
+    col = col[length(col)], border = NA
+    )
+    polygon(c(0, 0, 1 / 2, 1, 1, 1 / 2), c(
+      levels[1], levels[length(levels)], levels[length(levels)] + dl,
+      levels[length(levels)], levels[1], levels[1] - dl
+    ), border = "black", lwd = 2)
+    ylim0 <- range(levels)
+    prettyspecial <- pretty(ylim0)
+    prettyspecial <- prettyspecial[prettyspecial <= max(ylim0) &
+      prettyspecial >= min(ylim0)]
+    axis(4,
+      las = 1, cex.axis = cex.colorbar, at = prettyspecial,
+      labels = prettyspecial, ...
+    )
+  } else {
+    box()
+    axis(4, las = 1, cex.axis = cex.colorbar, ...)
+  }
+
+  # box, axis and leged
+  mtext(colorbar.label, line = line.label, side = 4, cex = cex.label, ...)
+
+  # resetting properties for starting a new plot (mfrow style)
+  par(old.par)
+  par(mfg = mfg.save, new = FALSE)
+  invisible()
+}
+
+cdo <- function(command, args="", input="", options="", output="") {
+  if (args != "") args <- paste0(",", args)
+  argstr <- paste0(options, " ", command, args, " ", input, " ", output)
```

Comment:
The input and output files need to be quoted, in case there are e.g. spaces or other special characters somewhere in the path.
```suggestion
  argstr <- paste0(options, " ", command, args, " '", input, "' '", output, "'")
```